### PR TITLE
MULE-19921: Redelivery Policy: infinite loop with blank message ID in a source configured with transactions

### PR DESCRIPTION
**Pull request by rbourbonnavarro**

### CODE REVIEW

File: modules/core-components/src/main/java/org/mule/runtime/core/internal/processor/IdempotentRedeliveryPolicy.java [hunk 1]
```diff
@@ -263,6 +265,11 @@ public CoreEvent process(CoreEvent event) throws MuleException {
         incrementCounter(messageId, createMessagingException(event, ex));
         throw ex;
       }
+    } catch (ObjectStoreException e) {
```
Comment (elrodro83) [inline modules/core-components/src/main/java/org/mule/runtime/core/internal/processor/IdempotentRedeliveryPolicy.java:268]: what if there was another problem in the object store not related to the key being null?

**Review by elrodro83 — CHANGES_REQUESTED** [review]

**Review by elrodro83 — APPROVED** [review]

**Comment by rbourbonnavarro** [comment]
--validate

**Comment by rbourbonnavarro** [comment]
--validate

**Comment by rbourbonnavarro** [comment]
--validate

**Pull request merged**